internal/tiltfile/k8s/locators.go

```diff
@@ -0,0 +1,117 @@
+package k8s
+
+import (
+	"fmt"
+
+	"go.starlark.net/starlark"
+
+	"github.com/tilt-dev/tilt/internal/k8s"
+	"github.com/tilt-dev/tilt/internal/tiltfile/value"
+)
+
+// Deserializing locators from starlark values.
+type JSONPathImageLocatorListSpec struct {
+	Specs []JSONPathImageLocatorSpec
+}
+
+func (s JSONPathImageLocatorListSpec) IsEmpty() bool {
+	return len(s.Specs) == 0
+}
+
+func (s *JSONPathImageLocatorListSpec) Unpack(v starlark.Value) error {
+	list := value.ValueOrSequenceToSlice(v)
+	for _, item := range list {
+		spec := JSONPathImageLocatorSpec{}
+		err := spec.Unpack(item)
+		if err != nil {
+			return err
+		}
+		s.Specs = append(s.Specs, spec)
+	}
+	return nil
+}
+
+func (s JSONPathImageLocatorListSpec) ToImageLocators(selector k8s.ObjectSelector) ([]k8s.ImageLocator, error) {
+	result := []k8s.ImageLocator{}
+	for _, spec := range s.Specs {
+		locator, err := spec.ToImageLocator(selector)
+		if err != nil {
+			return nil, err
+		}
+		result = append(result, locator)
+	}
+	return result, nil
+}
+
+type JSONPathImageLocatorSpec struct {
+	jsonPath string
+}
+
+func (s *JSONPathImageLocatorSpec) badValueErr(v starlark.Value) error {
+	return fmt.Errorf("Expected map of the form {'json_path': ..., 'repo_field': ..., 'tag_field': ...}. Actual: %s", v)
+}
+
+func (s *JSONPathImageLocatorSpec) Unpack(v starlark.Value) error {
+	var ok bool
+	s.jsonPath, ok = starlark.AsString(v)
+	if !ok {
+		return fmt.Errorf("Expected string, got: %s", v)
```

Comment:
if we're expressing the expected value as a type, it seems like we should use %T for the observed value (plus it'd be annoying if `v`'s `.String()` didn't make it obviously not-a-string, resulting in a message like "Expected string, got: foobar")

Reply:
done (though used the starlark Type() instead of the Go type